source/Engine.cpp

```diff
@@ -1488,7 +1490,17 @@ void Engine::CalculateStep()
 	Point newCenterVelocity;
 	if(flagship)
 	{
-		newCenter = flagship->Position();
+		if(flagship->IsThrusting())
+			offsetMultiplier = offsetMultiplier >= 1. ? 1.
+				: offsetMultiplier + offsetChange;
+		else
+			offsetMultiplier = offsetMultiplier <= 0. ? 0.
+				: offsetMultiplier - offsetChange;
+		double temp = offsetMultiplier < 0.5 ? 4 * pow(offsetMultiplier, 3) : 1 - pow((-2 * offsetMultiplier) + 2, 3) / 2;
+		offset = flagship->Velocity() * 30. * temp;
+		newCenter = Preferences::CameraAcceleration() == "on" ?
+			flagship->Position() + offset : Preferences::CameraAcceleration() == "reversed" ?
+			flagship->Position() - offset : flagship->Position();
```

Comment:
Instead of storing offset, why not store the `newCenter` as a data object? So that you can assign the new center instead of doing the double ternary thingy twice.

Reply:
will do this

Reply:
I am correcting myself, center and newCenter have to stay separated.
We would have done this earlier if it would work and it wont work.

Reply:
> I am correcting myself, center and newCenter have to stay separated.

Separated yes, but I mean storing `newCenter` as a data member so that it can be used to update `center` in Step.

Reply:
ok

Reply:
done